make dev_setup not working

every time I run make dev_setup I get this 

docker tag ghcr.io/ix/sandbox:351deee50d1620b5f0d4551ce88cba95 ghcr.io/ix/sandbox:latest
touch .sentinel/image
docker-compose run --rm sandbox ./manage.py graphql_schema --out ./frontend/schema.graphql
Dev Containers CLI: RPC pipe not configured. Message: {"args":["docker-credential-helper","get"],"stdin":"https://index.docker.io/v1/"}
Dev Containers CLI: RPC pipe not configured. Message: {"args":["docker-credential-helper","get"],"stdin":"https://index.docker.io/v1/"}
[+] Running 0/0
 ⠋ redis Pulling                                                                                                                                              0.0s
 ⠋ db Pulling                                                                                                                                                 0.0s
error getting credentials - err: exit status 255, out: ``
make: *** [Makefile:73: graphene_to_graphql] Error 1


I tried again from inside VScode and got a little bit different error

Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "./manage.py": stat ./manage.py: no such file or directory: unknown
make: *** [Makefile:73: graphene_to_graphql] Error 1

I found this issue in manage.py as well

Import "django.core.management" could not be resolved from source Pylance(reportMissingModuleSource) [Ln 11, Col 14|

I've never seen that error before, Iappears to be related to how your docker is configured. Do you have credential helpers enabled?

Does this drop you into a bash shell?
`docker run -it ghcr.io/ix/sandbox:latest /bin/bash` 

> Error response from daemon: failed to create shim task: OCI runtime create failed: runc create failed: unable to start container process: exec: "./manage.py": stat ./manage.py: no such file or directory: unknown
make: *** [Makefile:73: graphene_to_graphql] Error 1

This is probably from starting in the wrong working directory.  It should be `/var/app`

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -44,6 +44,18 @@ jobs:
           path: docker-image.tar
           key: ${{ steps.load_image_url.outputs.image-url }}
 
+  dev_image:
+    needs: build
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run pytest
+        uses: ./.github/actions/run-docker
+        with:
+          # build frontend
+          # For now just testing that frontend builds
+          command: make frontend
+
   pytest:
     needs: build
     runs-on: ubuntu-latest

--- a/Dockerfile
+++ b/Dockerfile
@@ -36,9 +36,7 @@ ENV PATH $PATH:$NODE_MODULES_BIN
 
 # NPM package installs
 RUN echo "[$NPM_DIR]"
-WORKDIR $NPM_DIR
 COPY package.json $NPM_DIR
-RUN npm install
 
 ENV WEBPACK_OUTPUT=/var/compiled-static
 

--- a/Makefile
+++ b/Makefile
@@ -55,7 +55,12 @@ image: ${IMAGE_SENTINEL}
 
 # full frontend build
 .PHONY: frontend
-frontend: compose graphene_to_graphql compile_relay webpack
+frontend: compose npm_install graphene_to_graphql compile_relay webpack
+
+# install npm packages
+.PHONY: npm_install
+npm_install: compose
+	docker-compose run --rm sandbox npm install
 
 # compile javascript
 .PHONY: webpack

--- a/docker-compose.yml
+++ b/docker-compose.yml
@@ -34,7 +34,7 @@ services:
       - redis
     volumes:
       - .:/var/app
-      #- .node_modules:/var/npm/node_modules
+      - .node_modules:/var/npm/node_modules
       - ./bin:/usr/bin/ix
       - .bash_profile:/root/.bash_profile
     env_file:

--- a/frontend/components/SliderInput.js
+++ b/frontend/components/SliderInput.js
@@ -1,0 +1,47 @@
+import React, { useState } from "react";
+import {
+  Box,
+  Flex,
+  Input,
+  Slider,
+  SliderFilledTrack,
+  SliderThumb,
+  SliderTrack,
+} from "@chakra-ui/react";
+
+const SliderInput = ({ min, max, defaultValue, onChange }) => {
+  const [value, setValue] = useState(defaultValue);
+
+  const handleChange = (newValue) => {
+    setValue(newValue);
+    onChange(newValue);
+  };
+
+  return (
+    <Flex alignItems="center" justifyContent="space-between">
+      <Slider
+        aria-label="slider"
+        min={min}
+        max={max}
+        value={value}
+        onChange={handleChange}
+      >
+        <SliderTrack>
+          <SliderFilledTrack />
+        </SliderTrack>
+        <SliderThumb />
+      </Slider>
+      <Box width="100px" marginLeft="1rem">
+        <Input
+          type="number"
+          value={value}
+          min={min}
+          max={max}
+          onChange={(e) => handleChange(e.target.value)}
+        />
+      </Box>
+    </Flex>
+  );
+};
+
+export default SliderInput;

--- a/ix/server/settings.py
+++ b/ix/server/settings.py
@@ -27,6 +27,7 @@ SECRET_KEY = os.environ["DJANGO_SECRET_KEY"]
 DEBUG = True
 
 ALLOWED_HOSTS = [
+    "localhost",
     "0.0.0.0",
     "127.0.0.1",
 ]

